packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts

```diff
@@ -120,7 +121,11 @@ export class PipelineBuilder {
     namespace: string,
     options: PreviewOptions
   ): Promise<Document[]> {
-    const pipeline = this.getPipelineFromSource();
+    // For preview we ignore $out/$merge stage.
+    const pipeline = [...this.getPipelineFromSource()];
+    if (isLastStageOutputStage(pipeline)) {
+      pipeline.splice(pipeline.length - 1, 1);
+    }
```

Comment:
I woulds still put it behind an option: we need to be explicit that this happens because it's still only text editor that should do this, if this somehow starts affecting stage editor, this should throw